cmake/protobuf_function.cmake

```diff
@@ -143,6 +143,7 @@ function(onnxruntime_protobuf_generate)
   endif()
   if(onnxruntime_protobuf_generate_TARGET)
     target_sources(${onnxruntime_protobuf_generate_TARGET} PRIVATE ${_generated_srcs_all})
+    target_link_libraries(${onnxruntime_protobuf_generate_TARGET} PUBLIC protobuf::libprotobuf)
```

Comment:
This is a pseudo target . Nobody would link to it.

Reply:
By doing this the target will carry protobuf dependency.
It avoids the non-intuitive injection of protobuf to those final targets who doesn't reference protobuf directly.